### PR TITLE
coap_client: propagate response errors to sychronous API calls

### PR DESCRIPTION
**Pull request by mniestroj**
So far there was no way to pass error codes (besides timeout) from
synchronous (CoAP) requests. Add `enum golioth_status` to the
`golioth_coap_request_msg_t` to make it possible to return non-success
status codes.

When response from server was received with error code, synchronous API
calls were still retuning `GOLIOTH_OK` status.

Propagate responses with error codes to the application as error status
`GOLIOTH_ERR_*`.

### CODE REVIEW

File: src/coap_client_zephyr.c [hunk 1]
```diff
@@ -178,6 +204,8 @@ static int golioth_coap_cb(struct golioth_req_rsp *rsp)
     {
         if (req->request_complete_event)
         {
+            *req->status = golioth_err_to_status(rsp->err);
```
Comment (sam-golioth) [inline src/coap_client_zephyr.c:207]: Do we need to make a similar change for libcoap?

Comment (mniestroj) [reply src/coap_client_zephyr.c:207]: Yes, that should be the case. Though, I am not sure yet how the implementation would look like.

Comment (sam-golioth) [reply src/coap_client_zephyr.c:207]: Ok, let's make the change for both platforms at the same time.

Comment (mniestroj) [reply src/coap_client_zephyr.c:207]: This PR was updated to cover all platforms, so please take another look @sam-golioth.

**Review by sam-golioth — APPROVED** [review]

**Pull request merged**